reaktive/src/commonTest/kotlin/com/badoo/reaktive/observable/TakeObservableTests.kt

```diff
@@ -0,0 +1,109 @@
+package com.badoo.reaktive.observable
+
+import com.badoo.reaktive.test.base.assertDisposed
+import com.badoo.reaktive.test.base.assertError
+import com.badoo.reaktive.test.base.assertNotError
+import com.badoo.reaktive.test.observable.*
+import kotlin.test.Test
+import kotlin.test.assertFailsWith
+
+class TakeObservableTests :
+    ObservableToObservableTests by ObservableToObservableTests<Unit>({ take(1) }) {
+
+    private val upstream = TestObservable<Int?>()
+    val observer = upstream.take(10).test()
+
+    @Test
+    fun completes_WHEN_limit_of_1_is_reached() {
+        val observer = upstream.take(1).test()
+
+        upstream.onNext(42)
+
+        observer.assertComplete()
+        observer.assertNotError()
+    }
+
+    @Test
+    fun completes_WHEN_limit_of_2_is_reached() {
+        val observer = upstream.take(2).test()
+
+        upstream.onNext(42)
+        observer.assertNotComplete()
+
+        upstream.onNext(84)
+        observer.assertComplete()
+        observer.assertNotError()
+    }
+
+    @Test
+    fun completes_WHEN_upstream_completed() {
+        upstream.onComplete()
+
+        observer.assertComplete()
+    }
+
+    @Test
+    fun produces_same_error_WHEN_upstream_produced_error() {
+        val error = Exception()
+
+        upstream.onError(error)
+
+        observer.assertError(error)
+    }
+
+    @Test
+    fun disposes_WHEN_upstream_disposed() {
```

Comment:
@arkivanov Please check this test, don't see any problems in code.

Reply:
The test is invalid. You should not dispose upstream manually, operators are responsible for it.
In this case `Disposable` passed to downstream `Observer` should be disposed when: 

1. Upstream produced a terminal event (`Complete` or `Error`)
2. Upstream produced a required amount of values

The first case should be covered in basic tests (`ObservableToObservableTests` etc.), it is not now, filed issue #195.
The second case should be covered here, like `disposes_disposable_WHEN_values_limit_is_reached`.

Reply:
> The first case should be covered in basic tests (ObservableToObservableTests etc.), it is not now, filed issue #195.

Sounds good!